### ISSUE TITLE
Visualizer not reacting to sound

### ISSUE THREAD

**Issue by treierxyz** [univrsal/spectralizer#21]
Can't get the visualizer to work.
Logs: [2020-05-24 23-19-54.txt](https://github.com/univrsal/spectralizer/files/4674341/2020-05-24.23-19-54.txt)

![Screenshot_20200524_231740](https://user-images.githubusercontent.com/33978534/82764063-e06d1080-9e14-11ea-9a5f-25aa98bbe6f0.png)

**Comment by univrsal** [comment]
Does it work with other audio sources? Maybe try changing back and forth in between audio sources

**Comment by treierxyz** [comment]
Changing audio sources didn't work.

**Comment by univrsal** [comment]
Can you try adding a media or vlc source and see if it works with those?

**Comment by treierxyz** [comment]
I will try tomorrow.

**Comment by treierxyz** [comment]
Media and VLC Source work fine. Also tried JACK Input Client, but that didn't work.

**Comment by treierxyz** [comment]
Also Audio Capture Device (ALSA) works.

**Comment by univrsal** [comment]
I really don't know what's causing this.

**Comment by treierxyz** [comment]
This actually started a while ago but I didn't get around to making a report and then I didn't use OBS for a while so I forgot about it.

**Comment by treierxyz** [comment]
Still having this problem, but a fresh VM had no problems, so it must be something with my OS installation. I could try to find out what is causing this but it's just much simpler to reinstall the OS, which I might probably do by the end of the year.

**Comment by univrsal** [comment]
I really can't help you with this much, I haven't had issues with either JACK or Pulseaudio. I'm rewriting the spectrum code (or at least have started) so maybe that'll fix it.

**Comment by treierxyz** [comment]
I reinstalled my OS and that didn't help, it's a problem with Pulseaudio and JACK.
I can get the spectralizer working by:
1) opening OBS
2) restarting JACK
3) setting desktop audio to Default

I have no problems when Pulseaudio only is used, but Pulse and JACK together don't mix well. It might also depend on if you use JACK1 or JACK2, which of I am using the latter.

But if you're already working on a rewrite, then I think this bug doesn't matter anymore.

**Comment by mikelupe** [comment]
Hello,
I'm having the same issue. A local media file is handled correctly, but any routed Jack source first shortly triggers some signal being displayed, and then basically immediately mutes.
I'm on Debian 11 btw. and I'm routing with Jack from MIXXX to OBS.

Do you have any new insights about this? Thanks. (great plugin)

**Comment by univrsal** [comment]
nope

**Comment by Hitrate** [comment]
Yes, no reaction to any audio. What a bummer. Hopefully you fix the app someday.

**Comment by remixmabix** [comment]
Same here

**Comment by LeonHartwig** [comment]
Jack users: as a test, try increasing your frames/period setting in your Jack setup. Make sure to restart Jack for the change to take effect. I normally use 256 and the visualizer doesn't react and spams the obs --verbose log with 'No Data in circle buffer' messages. However, if I configure Jack for 512 or 1024 frames/period, then the visualizer works. Could be some faulty assumptions in the visualizer regarding how much audio data to expect (and when) from an audio source.

**Comment by remixmabix** [comment]
Leon is on point, strange behaviour also because with Ardour if I go change the frames period from 512 to 64 it still visualize.

But if I change source again then it makes an error. Seems like the problem is in the initialisation of the plugin.

Thank you

**Comment by lcalder** [comment]
If I set my asio device to 512 samples it works fine. If I change the sample size to anything less than 512 samples, it will not react at all even though OBS audio meters show sound. If I switch back to 512 samples, spectralizer springs back to life.

**Comment by treierxyz** [comment]
> Jack users: as a test, try increasing your frames/period setting in your Jack setup. Make sure to restart Jack for the change to take effect. I normally use 256 and the visualizer doesn't react and spams the obs --verbose log with 'No Data in circle buffer' messages. However, if I configure Jack for 512 or 1024 frames/period, then the visualizer works. Could be some faulty assumptions in the visualizer regarding how much audio data to expect (and when) from an audio source.

Sorry, but sadly changing buffer size or periods/buffer didn't help, still no response. Sample rate also didn't have any effect. I don't get any "No Data in circle buffer" errors in logs so that could be a different issue?

**Comment by TheBlackParrot** [comment]
Getting this issue with Pipewire's JACK interface on Linux as well. I start OBS and the bars are always full with the Monstercat filter, and empty with any other filter. Never can get it to react.
I get similar issues on the Windows side too, my old layout had multiple Spectralizer sources and there was usually 1 out of the 3 that would initialize with full bars and never react. Swapping scene collections back and forth would usually re-init them.

**Comment by void-ghast** [comment]
This issue with the sample rate less than 512 not showing up is a huge bummer. Shouldn't be too difficult to fix on the developer's end right?
